lib/Windowing/InViewList.js

```diff
@@ -0,0 +1,67 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { func, node, number } from 'prop-types';
+import { WindowingContext } from './WindowingProvider';
+
+function InViewList({ items, offset, children, itemHeight }) {
+  const [renderIndexes, setRenderIndexes] = useState({});
+  const { scrollTop, height } = useContext(WindowingContext);
+
+  const subtractFromStartIndex = index =>
+    index - offset <= 0 ? 0 : index - offset;
+
+  const addToEndIndex = index =>
+    index + offset >= items.length ? items.length : index + offset;
+
+  const getRenderIndexes = () => {
+    const startBoundary = scrollTop;
+    const endBoundary = startBoundary + height;
+
+    const startIndex = Math.round(startBoundary / itemHeight);
+    const endIndex = Math.round(endBoundary / itemHeight);
+
+    const startWithOffset = subtractFromStartIndex(startIndex);
+    const endWithOffset = addToEndIndex(endIndex);
+
+    setRenderIndexes({
+      start: startWithOffset,
+      end: endWithOffset
+    });
+  };
+
+  useEffect(() => {
+    getRenderIndexes();
```

Comment:
updates the render indexes every time the `height` or `scroll` is updated which in the provider is updates on `resize` and `scroll`.